packages/core/test/Cardano/types/Strings/util.test.ts

```diff
@@ -0,0 +1,59 @@
+import { InvalidStringError, assertIsBech32WithPrefix, assertIsHexString } from '../../../../src/Cardano';
+
+describe('Cardano/types/Strings/util', () => {
+  describe('assertIsBech32WithPrefix', () => {
+    it('does not throw when asserting a valid bech32', () => {
+      expect(() =>
+        assertIsBech32WithPrefix('pool1ujcu3myfg9wwvdyh2ks653954lamtufy3lefjs73jrr327q53j4', 'pool')
+      ).not.toThrow();
+      expect(() =>
+        assertIsBech32WithPrefix('pool1ujcu3myfg9wwvdyh2ks653954lamtufy3lefjs73jrr327q53j4', 'pool', 45)
+      ).not.toThrow();
+    });
+
+    it('throws when decoded words length does not match the last argument', () => {
+      expect(() =>
+        assertIsBech32WithPrefix('pool1ujcu3myfg9wwvdyh2ks653954lamtufy3lefjs73jrr327q53j4', 'pool', 55)
+      ).toThrowError(InvalidStringError);
+    });
+
+    it('throws when asserting an invalid bech32 (too short)', () => {
+      expect(() => assertIsBech32WithPrefix('pool1', 'pool')).toThrowError(InvalidStringError);
+    });
+
+    it('throws when asserting an empty string', () => {
+      expect(() => assertIsBech32WithPrefix('', '')).toThrowError(InvalidStringError);
+    });
+
+    it('throws when decoded prefix does not match the expected prefix', () => {
```

Comment:
This is throwing at an unexpected execution point. Making the assertion more specific will produce a failing test

![image](https://user-images.githubusercontent.com/303881/143188425-528f8f47-17bf-4146-8cbe-2f2d57a14151.png)